gateway/mw_auth_key.go

```diff
@@ -102,19 +103,28 @@ func (k *AuthKey) ProcessRequest(_ http.ResponseWriter, r *http.Request, _ inter
 		return errorAndStatusCode(ErrAuthAuthorizationFieldMissing)
 	}
 
-	session, keyExists = k.CheckSessionAndIdentityForValidKey(&key, r)
+	session, keyExists = k.CheckSessionAndIdentityForValidKey(key, r)
+	key = session.KeyID
 	if !keyExists {
 		// fallback to search by cert
-		session, keyExists = k.CheckSessionAndIdentityForValidKey(&certHash, r)
+		session, keyExists = k.CheckSessionAndIdentityForValidKey(certHash, r)
+		certHash = session.KeyID
 		if !keyExists {
 			return k.reportInvalidKey(key, r, MsgNonExistentKey, ErrAuthKeyNotFound)
 		}
 	}
 
 	if authConfig.UseCertificate {
 		certID := session.OrgID + certHash
-		if _, err := CertificateManager.GetRaw(certID); err != nil {
-			return k.reportInvalidKey(key, r, MsgNonExistentCert, ErrAuthCertNotFound)
+		_, err := CertificateManager.GetRaw(certID)
+		if err != nil {
+			// Try alternative approach:
+			id := storage.TokenID(session.KeyID)
```

Comment:
I do not see how it can be related to your changes. 
certID is always `session.OrgID + certHash`. 
It should have nothing to do wit  keyIDs or similar.

Reply:
In previous changes we had the following:
```go
        session, keyExists = k.CheckSessionAndIdentityForValidKey(key, r)
        key = session.KeyID
        if !keyExists {
                // fallback to search by cert
                session, keyExists = k.CheckSessionAndIdentityForValidKey(certHash, r)
                certHash = session.KeyID
                if !keyExists {
                        return k.reportInvalidKey(key, r, MsgNonExistentKey, ErrAuthKeyNotFound)
                }
        }

        if authConfig.UseCertificate {
                certID := session.OrgID + certHash
                _, err := CertificateManager.GetRaw(certID)
                if err != nil {

```
We reviewed the tests too and they were broken, after adding this `GetRaw` workaround it was fixed.

Reply:
Just for the records, without this change the test `TestHashKeyFunctionChanged/client_certificate` fails...we need to check why it passed before